metadata-models/src/main/pegasus/com/linkedin/assertion/AssertionInfo.pdl

```diff
@@ -14,22 +14,12 @@ record AssertionInfo includes CustomProperties, ExternalReference {
     * Type of assertion. Assertion types can evolve to span Datasets, Flows (Pipelines), Models, Features etc.
     */
     type: enum AssertionType {
-      // When present, then DatasetAssertionInfo elements are filled out
+      // A single-dataset assertion. When this is the value, the datasetAssertion field will be populated.
       DATASET
     }
 
     /**
     * Dataset Assertion information when type is DATASET
     */
     datasetAssertion: optional DatasetAssertionInfo
-
-    /*
-    * Logic for the assertion as expressed in the native assertion language. Code fragments, query strings, etc.
-    */
-    assertionLogic: optional string
```

Comment:
Just to confirm, removing optional fields is backward compatible?

Reply:
It's not - if you were previously writing this field then things would break. We don't expect anyone is using these directly since there's no docs/integrations yet